elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchIndexManager.java

```diff
@@ -267,7 +270,12 @@ private void waitForIndexCreation() {
 		Health health = new Health( healthBuilder ) {
 			@Override
 			protected String buildURI() {
-				return super.buildURI() + actualIndexName;
+				try {
+					return super.buildURI() + URLEncoder.encode(actualIndexName, AbstractAction.CHARSET);
+				}
+				catch (UnsupportedEncodingException e) {
+					throw new AssertionFailure("Unexpectedly unsupported charset", e);
+				}
```

Comment:
I'm wondering why do we do the index name encoding only in this case? Do we need a follow-up issue to URL-encode the index name only once (at initialization of the IndexManager) and then make sure we use that consistently?

Secondarily.. this code isn't respecting the formatting conventions around for whitespacing. Wondering how you managed to fool checkstyle :)